data_steward/cdr_cleaner/cleaning_rules/remove_non_matching_participant.py

```diff
@@ -93,33 +104,37 @@ def get_list_non_match_participants(project_id, validation_dataset_id, hpo_id):
 
     # get the the hpo specific <hpo_id>_identity_match
     identity_match_table = bq_utils.get_table_id(hpo_id, IDENTITY_MATCH)
+    result = []
+    if exist_identity_match(identity_match_table, validation_dataset_id):
+        non_match_participants_query = get_non_match_participant_query(
+            project_id, validation_dataset_id, identity_match_table)
 
-    non_match_participants_query = get_non_match_participant_query(
-        project_id, validation_dataset_id, identity_match_table)
-
-    try:
-        LOGGER.info(
-            'Identifying non-match participants in {dataset_id}.{identity_match_table}'
-            .format(dataset_id=validation_dataset_id,
-                    identity_match_table=identity_match_table))
-
-        results = bq_utils.query(q=non_match_participants_query)
-
-    except (oauth2client.client.HttpAccessTokenRefreshError,
-            googleapiclient.errors.HttpError) as exp:
-
-        LOGGER.exception('Could not execute the query \n{query}'.format(
-            query=non_match_participants_query))
-        raise exp
-
-    # wait for job to finish
-    query_job_id = results['jobReference']['jobId']
-    incomplete_jobs = bq_utils.wait_on_jobs([query_job_id])
-    if incomplete_jobs:
-        raise bq_utils.BigQueryJobWaitError(incomplete_jobs)
-
-    # return the person_ids only
-    return [row[PERSON_ID_FIELD] for row in bq_utils.response2rows(results)]
+        try:
+            LOGGER.info(
+                'Identifying non-match participants in {dataset_id}.{identity_match_table}'
+                .format(dataset_id=validation_dataset_id,
```

Comment:
Is this showing up in git diff because of the indent? You could skip the indenting by returning an empty list in the conditional and changing it to `if not exists`.

Reply:
That is not just the indent, the whole chunk of code is moved to process out of the loop.